sys/include/net/gnrc/lwmac/hdr.h

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2015 Daniel Krebs
+ *               2016 INRIA
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     net_lwmac
+ * @{
+ *
+ * @file
+ * @brief       Header definition LWMAC
+ * @internal
+ * @author      Daniel Krebs <github@daniel-krebs.net>
+ * @author      Shuguo Zhuo  <shuguo.zhuo@inria.fr>
+ */
+
+#ifndef GNRC_LWMAC_HDR_H_
+#define GNRC_LWMAC_HDR_H_
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#include "net/ieee802154.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   lwMAC internal L2 address structure
+ */
+typedef struct {
+    uint8_t addr[IEEE802154_LONG_ADDRESS_LEN]; /**< address of node */
+    uint8_t len;                               /**< address */
+} l2_addr_t;
+
+/**
+ * @brief Static initializer for l2_addr_t.
+ */
+#define LWMAC_L2_ADDR_INIT      { { 0 }, 0 }
+
+/**
+ * @brief   lwMAC frame types
+ */
+typedef enum {
+    FRAMETYPE_WR = 1,
+    FRAMETYPE_WA,
+    FRAMETYPE_DATA,
+    FRAMETYPE_DATA_PENDING,
+    FRAMETYPE_BROADCAST,
+} lwmac_frame_type_t;
+
+/**
+ * @brief   lwMAC header
+ */
+typedef struct __attribute__((packed)) {
+    lwmac_frame_type_t type; /**< type of frame */
```

Comment:
I assume this struct is gonna go over the wire?

- please use an explicit type (not enum) for ```lwmac_frame_type_t```. The size of an enum is dependent on architecture and compiler(-options).
- no need to "pack" a struct with only one field